src/internal/connector/support/m365Transform.go

```diff
@@ -346,7 +384,38 @@ func sanitizeEvent(orig models.Eventable) (models.Eventable, error) {
 	newEvent.SetIsDraft(nil)
 	newEvent.SetAdditionalData(orig.GetAdditionalData())
 
-	attached := orig.GetAttachments()
+	attachments, err := sanitizeAttachments(orig.GetAttachments())
+	if err != nil {
+		return nil, err
+	}
+
+	newEvent.SetAttachments(attachments)
```

Comment:
@dadams39 - are events with nested attachments handled correctly? If not - we will likely need to update this logic also to mimic below.